externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/session/SparkSessionImpl.scala

```diff
@@ -53,6 +54,7 @@ class SparkSessionImpl(
       case (key, value) => setModifiableConfig(key, value)
     }
     KDFRegistry.registerAll(spark)
+    sessionEvent.serverIp = serverIpAddress
```

Comment:
inside `SessionEvent.apply`

Reply:
> inside `SessionEvent.apply`

change `SessionEvent.apply(session: Session)` to `SessionEvent.apply(session: SparkSessionImpl)`, or use `session.asInstanceOf[SparkSessionImpl].serverIpAddress` in SessionEvent, which one is better?

Reply:
I prefer the first option.

Reply:
> I prefer the first option.

Thanks @ulysses-you , I changed it to that.